starters/basic-starter/components/media--image.tsx

```diff
@@ -17,26 +18,56 @@ export function MediaImage({
   height,
   priority,
   sizes,
+  imageStyle,
   ...props
 }: MediaImageProps) {
   const image = media?.image;
 
   if (!image) {
     return null;
   }
+  let sizeProps;
+  let srcURL;
+  let useImageStyle = false;
 
-  const sizeProps =
-    layout === 'fill'
-      ? null
-      : {
-          width: width || image.resourceIdObjMeta.width,
-          height: height || image.resourceIdObjMeta.height,
+  // Use the image style to render an image if specified.
+  if (imageStyle) {
+    if (image.links) {
+      if (image.links[imageStyle]) {
+        const imageStyleSource = image.links[imageStyle];
+        sizeProps = {
+          width: width || imageStyleSource.meta.linkParams.width,
+          height: height || imageStyleSource.meta.linkParams.height,
         };
+        srcURL = imageStyleSource.href;
+        useImageStyle = true;
+      } else {
+        console.warn(
+          `${imageStyle} image style does not exist. You must configure the consumer to include the image style to be used by this application.`,
+        );
+      }
+    } else {
+      console.warn(
+        'Image styles not found. The consumer_image_styles module must be installed and enabled for the consumer of this application.',
+      );
+    }
+  }
+  if (!useImageStyle) {
```

Comment:
I think this code could be simplified slightly; we could set these values at first always and then override them if image style is used.